dao/src/test/java/org/thingsboard/server/dao/sql/query/DefaultQueryLogComponentTest.java

```diff
@@ -0,0 +1,128 @@
+
+package org.thingsboard.server.dao.sql.query;
+
+import com.datastax.oss.driver.api.core.uuid.Uuids;
+import org.junit.jupiter.api.BeforeEach;
+import org.junit.jupiter.api.Test;
+import org.junit.runner.RunWith;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.boot.test.context.SpringBootTest;
+import org.springframework.test.context.junit4.SpringRunner;
+import org.thingsboard.server.common.data.EntityType;
+import org.thingsboard.server.common.data.id.TenantId;
+
+import java.util.ArrayList;
+import java.util.List;
+import java.util.UUID;
+
+import static org.junit.Assert.assertEquals;
+
+@RunWith(SpringRunner.class)
+@SpringBootTest(classes = DefaultQueryLogComponent.class)
+class DefaultQueryLogComponentTest {
+
+    private TenantId tenantId;
+    private QueryContext ctx;
+
+    @Autowired
+    private DefaultQueryLogComponent queryLog;
+
+    @BeforeEach
+    void setUp() {
+        tenantId = new TenantId(Uuids.timeBased());
+        ctx = new QueryContext(new QuerySecurityContext(tenantId, null, EntityType.ALARM));
+    }
+
+    @Test
+    void substituteParametersInSqlString_StringType() {
+
+        String Name = "Mery";
+        String id = "ID_1";
+        String sql = "Select * from Table Where name = :name AND id = :id";
+        String sqlToUse = "Select * from Table Where name = 'Mery' AND id = 'ID_1'";
+
+        ctx.addStringParameter("name", Name);
+        ctx.addStringParameter("id", id);
+
+        String sqlToUse2 = queryLog.substituteParametersInSqlString(sql, ctx);
+        assertEquals(sqlToUse, sqlToUse2);
+    }
+
+    @Test
+    void substituteParametersInSqlString_DoubleLongType() {
+
+        double sum = 10.55;
```

Comment:
double with many 000000 may be printed as a scientific notation. I suggest you add a specific use case on this.
```
double myvalue = 0.00000021d;
```
```
2.1E-7
```